src/logic/map_objects/bob.cc

```diff
@@ -705,34 +706,41 @@ Vector2f Bob::calc_drawpos(const EditorGameBase& game,
 	const float triangle_w = kTriangleWidth * scale;
 	const float triangle_h = kTriangleHeight * scale;
 
+	bool bridge = false;
 	switch (walking_) {
 	case WALK_NW:
 		map.get_brn(end, &start);
 		spos.x += triangle_w / 2.f;
 		spos.y += triangle_h;
+		bridge = is_bridge_segment(end.field->road_southeast);
 		break;
 	case WALK_NE:
 		map.get_bln(end, &start);
 		spos.x -= triangle_w / 2.f;
 		spos.y += triangle_h;
+		bridge = is_bridge_segment(end.field->road_southwest);
 		break;
 	case WALK_W:
 		map.get_rn(end, &start);
 		spos.x += triangle_w;
+		bridge = is_bridge_segment(end.field->road_east);
 		break;
 	case WALK_E:
 		map.get_ln(end, &start);
 		spos.x -= triangle_w;
+		bridge = is_bridge_segment(start.field->road_east);
 		break;
 	case WALK_SW:
 		map.get_trn(end, &start);
 		spos.x += triangle_w / 2.f;
 		spos.y -= triangle_h;
+		bridge = is_bridge_segment(start.field->road_southwest);
 		break;
 	case WALK_SE:
 		map.get_tln(end, &start);
 		spos.x -= triangle_w / 2.f;
 		spos.y -= triangle_h;
+		bridge = is_bridge_segment(start.field->road_southeast);
```

Comment:
You can move this below `if (start.field) {` and get rid of the repetition.

Reply:
I don´t understand what you mean, this function gets six different parameters in the six cases?

Reply:
You're right, I overlooked this.